File: ejercicio1.py

class Alumno:
    def __init__(self, nombre, calificacion):
        self.nombre = nombre
        self.calificacion = calificacion

    def aprobado(self):
        return self.calificacion >= 6

alumnos = [
    Alumno("Ana", 9),
    Alumno("Luis", 5),
    Alumno("Sofia", 8)
]

for alumno in alumnos:
    print(alumno.nombre, alumno.calificacion)
    if alumno.aprobado():
        print("Aprobado")
    else:
        print("Reprobado")

File: ejercicio1.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Alumno {
public:
    string nombre;
    float calificacion;

    Alumno(string n, float c) {
        nombre = n;
        calificacion = c;
    }

    bool aprobado() {
        return calificacion >= 6;
    }
};

int main() {
    vector<Alumno> alumnos;
    alumnos.push_back(Alumno("Ana", 9));
    alumnos.push_back(Alumno("Luis", 5));
    alumnos.push_back(Alumno("Sofia", 8));

    for (int i = 0; i < alumnos.size(); i++) {
        cout << alumnos[i].nombre << ": " << alumnos[i].calificacion << endl;
        if (alumnos[i].aprobado()) {
            cout << "Aprobado" << endl;
        } else {
            cout << "Reprobado" << endl;
        }
    }

    return 0;
}
